src/main/java/seedu/modtrek/logic/parser/AddCommandParser.java

```diff
@@ -0,0 +1,60 @@
+package seedu.modtrek.logic.parser;
+
+import static seedu.modtrek.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.modtrek.logic.parser.CliSyntax.PREFIX_CODE;
+import static seedu.modtrek.logic.parser.CliSyntax.PREFIX_CREDIT;
+import static seedu.modtrek.logic.parser.CliSyntax.PREFIX_GRADE;
+import static seedu.modtrek.logic.parser.CliSyntax.PREFIX_SEMYEAR;
+import static seedu.modtrek.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import seedu.modtrek.logic.commands.AddCommand;
+import seedu.modtrek.logic.parser.exceptions.ParseException;
+import seedu.modtrek.model.module.Code;
+import seedu.modtrek.model.module.Credit;
+import seedu.modtrek.model.module.Grade;
+import seedu.modtrek.model.module.Module;
+import seedu.modtrek.model.module.SemYear;
+import seedu.modtrek.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new AddCommand object
+ */
+public class AddCommandParser implements Parser<AddCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_CODE, PREFIX_CREDIT, PREFIX_SEMYEAR, PREFIX_GRADE, PREFIX_TAG);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_CODE, PREFIX_CREDIT, PREFIX_SEMYEAR, PREFIX_GRADE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+        }
+
+        Code code = ParserUtil.parseCode(argMultimap.getValue(PREFIX_CODE).get());
+        Credit credit = ParserUtil.parseCredit(argMultimap.getValue(PREFIX_CREDIT).get());
+        SemYear semYear = ParserUtil.parseSemYear(argMultimap.getValue(PREFIX_SEMYEAR).get());
+        Grade grade = ParserUtil.parseGrade(argMultimap.getValue(PREFIX_GRADE).get());
```

Comment:
Just to check, grade can be null? I didn't set the model to check for non-null grade since the user can add a module without taking it yet. Will this affect the parser here?

Reply:
Also I left a bunch of valid grades to check for under `models/module/Grade.java`. Not sure if you have checked to deal with the validity of the grades.

Reply:
The validity of the grade is checked when a Grade object is instantiated using the grade array you guys wrote. I just changed the code to make grade optional(no grade is just an empty string) but I'm not letting it be null (to avoid NPE in the future I guess). Thanks for pointing that out.